File: spec/functional_test/fixtures/swift/cli_argumentparser/Tool.swift
import ArgumentParser

struct Tool: ParsableCommand {
    static let configuration = CommandConfiguration(
        commandName: "tool",
        subcommands: [Serve.self]
    )

    @Flag(name: .shortAndLong) var verbose = false
}

struct Serve: ParsableCommand {
    static let configuration = CommandConfiguration(commandName: "serve")

    @Option(name: .long) var port: Int = 8080
    @Argument var config: String
}

File: spec/functional_test/fixtures/lua/cli_argparse/tool.lua
local argparse = require("argparse")

local parser = argparse("tool", "My tool")
parser:flag("-v --verbose", "Verbose")
parser:option("-p --port", "Port")

local serve = parser:command("serve", "Start server")
serve:option("--host", "Host")
serve:argument("config", "Config file")

local token = os.getenv("API_TOKEN")
local first = arg[1]
print(token, first)
